Src/ReadBarometer.c

```diff
@@ -6,127 +17,244 @@
+/* Constants -----------------------------------------------------------------*/
 
-static const uint8_t ADC_D1_512_CONV_CMD = 0x42;
-static const uint8_t ADC_D2_512_CONV_CMD = 0x52;
-static const uint8_t ADC_READ_CMD = 0x00;
-static const uint8_t PROM_READ_SENS_CMD = 0xA2;
-static const uint8_t PROM_READ_OFF_CMD = 0xA4;
-static const uint8_t PROM_READ_TCS_CMD = 0xA6;
-static const uint8_t PROM_READ_TCO_CMD = 0xA8;
-static const uint8_t PROM_READ_TREF_CMD = 0xAA;
+static const uint8_t ADC_D1_512_CONV_CMD    = 0x42;
+static const uint8_t ADC_D2_512_CONV_CMD    = 0x52;
+static const uint8_t ADC_READ_CMD           = 0x00;
+static const uint8_t PROM_READ_SENS_CMD     = 0xA2;
+static const uint8_t PROM_READ_OFF_CMD      = 0xA4;
+static const uint8_t PROM_READ_TCS_CMD      = 0xA6;
+static const uint8_t PROM_READ_TCO_CMD      = 0xA8;
+static const uint8_t PROM_READ_TREF_CMD     = 0xAA;
 static const uint8_t PROM_READ_TEMPSENS_CMD = 0xAC;
-static const uint8_t RESET_CMD = 0x1E;
+static const uint8_t READ_BYTE_CMD          = 0x00;
+static const uint8_t RESET_CMD              = 0x1E;
+
+/* Variables -----------------------------------------------------------------*/
+
+/* Structs -------------------------------------------------------------------*/
+
+/* Prototypes ----------------------------------------------------------------*/
 
 uint16_t readCalibrationCoefficient(const uint8_t PROM_READ_CMD);
 
+/* Functions -----------------------------------------------------------------*/
+
+/**
+ * This function is to be used as a thread task that reads and updates
+ * pressure and temperature readings from the barometer.
+ */
 void readBarometerTask(void const* arg)
 {
-    BarometerData* data = (BarometerData*) arg;
-    uint32_t prevWakeTime = osKernelSysTick();
+    /**
+     * Variable Descriptions from MS5607-02BA03 Data Sheet:
+     *
+     * C1 (SENSt1)      - Pressure sensitivity
+     * C2 (OFFt1)       - Pressure offset
+     * C3 (TCS)         - Temperature coefficient of pressure sensitivity
+     * C4 (TCO)         - Temperature coefficient of pressure offset
+     * C5 (Tref)        - Reference temperature
+     * C6 (TEMPSENS)    - Temperature coefficient of the temperature
+     *
+     * D1   - Digital pressure value
+     * D2   - Digital temperature value
+     *
+     * dT   - Difference between actual and reference temperature
+     *          dT = D2 - Tref = D2 - (C5 * 2^8)
+     * TEMP - Actual temperature (-40...85°C with 0.01°C resolution)
+     *          TEMP = 20°C + (dT * TEMPSENS) = 2000 + (dT * C6)/2^23
+     * OFF  - Offset at actual temperature
+     *          OFF = OFFt1 + (TCO * dT) = (C2 * 2^17) + (C4 * dT)/2^6
+     * SENS - Sensitivity at actual temperature
+     *          SENS = SENSt1 + (TCS * dT) = (C1 * 2^16) + (C3 * dT)/2^7
+     * P    - Temperature compensated pressure (10...1200mbar with 0.01mbar resolution)
+     *          P = (D1 * SENS) - OFF = ((D1 * SENS)/2^21 - OFF)/2^15
+     */
```

Comment:
I think this section does make it easier to understand the code without using the datasheet. My only concern with this is if there is a mistake we do not notice, and then people reference the mistake instead of the datasheet. Maybe this is not a valid concern because we are reviewing it anyway.

Reply:
I think that's a valid concern, I mean I didn't read the datasheet haha.

Reply:
I've triple-checked that the information in this comment is correct. Hopefully it will make for a lot less window switching :stuck_out_tongue:

Reply:
Let's keep this comment in the code. I thought about it, and I think it will save us a lot of time in the future. I checked it over, and it matches the info on the datasheet.
I think adding comments like this to other files would be a good idea too.